packages/node/test/integration/secure-deposit.spec.ts

```diff
@@ -39,4 +43,125 @@ describe("Node method follows spec - deposit", () => {
       expect(freeBalanceState[key]).toEqual(One);
     }
   });
+
+  it("has the right balance for both parties after deposits of ERC20 tokens and ETH", async () => {
+    const multisigAddress = await createChannel(nodeA, nodeB);
+    const erc20ContractAddress = global["networkContext"]["DolphinCoin"];
+    const erc20Contract = new Contract(
+      erc20ContractAddress,
+      DolphinCoin.abi,
+      new JsonRpcProvider(global["ganacheURL"])
+    );
+
+    let depositReq = makeDepositRequest(
+      multisigAddress,
+      One,
+      erc20ContractAddress
+    );
+
+    expect(
+      await erc20Contract.functions.balanceOf(await nodeA.signerAddress())
+    ).toEqual(Zero);
+
+    let error;
+    try {
+      await nodeA.router.dispatch(depositReq);
+    } catch (e) {
+      error = e;
+    } finally {
+      expect(error).toEqual(
+        INSUFFICIENT_ERC20_FUNDS(await nodeA.signerAddress())
+      );
+    }
+
+    await transferERC20Tokens(await nodeA.signerAddress());
+    await transferERC20Tokens(await nodeB.signerAddress());
+
+    let preDepositBalance = await provider.getBalance(multisigAddress);
+    await nodeA.router.dispatch(depositReq);
+    await nodeB.router.dispatch(depositReq);
+
+    expect(await provider.getBalance(multisigAddress)).toEqual(
+      preDepositBalance
+    );
+    expect(await erc20Contract.functions.balanceOf(multisigAddress)).toEqual(
+      Two
+    );
+
+    await confirmEthAndERC20FreeBalances(
+      nodeA,
+      multisigAddress,
+      erc20ContractAddress
+    );
+    await confirmEthAndERC20FreeBalances(
+      nodeB,
+      multisigAddress,
+      erc20ContractAddress
+    );
+
+    // now deposits ETH
+
+    depositReq = makeDepositRequest(multisigAddress, One);
```

Comment:
Can you rename this helper method? Like constructDepositParams or something f

Reply:
This is a pretty accurate name for what it does actually. We're not interested in the params, we quite simply want to make a request to deposit.